### PR TITLE
API review updates

### PR DESCRIPTION
**Pull request by ajcvickers**
- Rename ConverterMappingHints.Override to WithOverride
- Rename the RelationalTypeMapping.Clone methods to sue With
- Rename ILoggingOptions.ShouldWarnForEnumType to ShouldWarnForStringEnumValueInJson
- Rename parameter to enumType
- Rename IClrPropertyGetter/Setter methods
- Make IReadOnlyProperty.GetElementType return IReadOnlyElementType
- Make NullTypeMapping pubternal
- Remove the SQLite HasSrid extension methods from primitive collection builders
- Remove IInternalEntry
- Use ITypeBase for ParameterBindingInfo constructor

### CODE REVIEW

File: src/EFCore.Relational/Storage/RelationalTypeMapping.cs [hunk 1]
```diff
@@ -443,35 +444,35 @@ public virtual RelationalTypeMapping Clone(int? precision, int? scale)
...
+    //             size: mappingInfo.Value.Size,
+    //             precision: mappingInfo.Value.Precision,
+    //             scale: mappingInfo.Value.Scale),
+    //         clrType,
+    //         converter,
+    //         comparer,
+    //         keyComparer,
+    //         providerValueComparer,
+    //         elementMapping,
+    //         jsonValueReaderWriter);
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Storage/RelationalTypeMapping.cs:475]: 👀

File: src/EFCore/Metadata/IConventionProperty.cs [hunk 2]
```diff
@@ -460,17 +460,27 @@ bool IsImplicitlyCreated()
...
+
     /// <summary>
     ///     Sets the configuration for elements of the primitive collection represented by this property.
     /// </summary>
-    /// <param name="primitiveCollection">If <see langword="true"/>, then this is a collection of primitive elements.</param>
+    /// <param name="elementType">If <see langword="true"/>, then the type mapping has an element type, otherwise it is removed.</param>
     /// <param name="fromDataAnnotation">Indicates whether the configuration was specified using a data annotation.</param>
     /// <returns>The configuration for the elements.</returns>
-    IElementType? ElementType(bool primitiveCollection, bool fromDataAnnotation = false);
+    IConventionElementType? SetElementType(bool elementType, bool fromDataAnnotation = false);
```
Comment (AndriySvyryd) [inline src/EFCore/Metadata/IConventionProperty.cs:475]: I'm not convinced this is the best name. `SetIsPrimitiveCollection` makes more sense to me.

**Review by AndriySvyryd — APPROVED** [review]

File: src/EFCore/Metadata/IConventionProperty.cs (hunk 2 above)
Comment (ajcvickers) [reply src/EFCore/Metadata/IConventionProperty.cs:475]: I plan to work on another PR today that allows an element type that is not the type of a primitive collection. This allows things like ranges to be mapped with an element type.

Comment (ajcvickers) [reply src/EFCore/Metadata/IConventionProperty.cs:475]: /cc @roji

Comment (AndriySvyryd) [reply src/EFCore/Metadata/IConventionProperty.cs:475]: Even so, this refers to the existence of an element type, not the element type itself.
`SetHasElementType`?

Comment (ajcvickers) [reply src/EFCore/Metadata/IConventionProperty.cs:475]: @AndriySvyryd See https://github.com/dotnet/efcore/pull/31677

**Comment by danmoseley** [comment]
Any changes in here (or other change between RC1 and RC2) that are breaking, we need to add to the efcore breaking change doc because RC1 was a go-live release.

The change in RelationalTypeMapping seems to be API breaking, eg..

**Comment by roji** [comment]
@danmoseley in an offline chat I summarized the different approach we have to customer-facing breaking changes and provider-facing ones, let's maybe have a chat about this if you want.

**Comment by danmoseley** [comment]
Ah interesting yes would like to learn, we can chat offline.

**Pull request merged**